control_plane/attachment_service/src/service.rs

```diff
@@ -505,7 +555,18 @@ impl Service {
             // Block the [`Service::startup_complete`] barrier until we're done
             let _completion = startup_completion;
 
-            startup_reconcile_this.startup_reconcile().await
+            // Block shutdown until we're done (we must respect self.cancel)
+            if let Ok(_gate) = startup_reconcile_this.gate.enter() {
+                startup_reconcile_this.startup_reconcile().await
```

Comment:
Both `startup_reconcile` and `background_reconcile` end up calling `maybe_reconcile` which enters the gate again. This is kind of weird, but I guess it's fine.